web-common/src/features/sources/sourceUtils.ts

```diff
@@ -37,6 +41,32 @@ export function compileCreateSourceYAML(
   return `${topLineComment}\n\ntype: "${connectorName}"\n` + compiledKeyValues;
 }
 
+function buildDuckDbQuery(path: string): string {
+  const extension = extractFileExtension(path as string);
+  if (extensionContainsParts(extension, [".csv", ".tsv", ".txt"])) {
+    return `select * from read_csv_auto('${path}')`;
```

Comment:
Let's be consistent with `read_json` and use `read_csv` with `auto_detect=true`? Nishant also [cites this spec in Notion](https://www.notion.so/rilldata/infer-types-for-untyped-sources-and-enable-correction-via-YAML-editor-851a7555d056497ba945dcb930348f6e?pvs=4#df6a1bebbf574045a7cd1be5d3ef0d80).

Nishant also has `ignore_errors=1, header=true`. Add those too?